packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -163,12 +206,53 @@ export class Table extends Construct {
     }
 
     public addPartitionKey(attribute: Attribute): this {
-        this.addKey(attribute.name, attribute.type, HASH_KEY_TYPE);
+        this.addKey(attribute, HASH_KEY_TYPE);
         return this;
     }
 
     public addSortKey(attribute: Attribute): this {
-        this.addKey(attribute.name, attribute.type, RANGE_KEY_TYPE);
+        this.addKey(attribute, RANGE_KEY_TYPE);
+        return this;
+    }
+
+    public addGlobalSecondaryIndex(props: SecondaryIndexProps): this {
+        if (this.globalSecondaryIndexes.length === 5) {
+            // https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html#limits-secondary-indexes
+            throw new RangeError('a maximum number of global secondary index per table is 5');
+        }
+
+        if (props.projectionType === ProjectionType.Include && !props.nonKeyAttributes) {
+            // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-projectionobject.html
```

Comment:
Thanks for including these links, that's fantastic!

Reply:
Thanks :)